app/models/project.rb

```diff
@@ -97,7 +95,7 @@ def init
         Rugged::Repository.init_at self.barerepopath, :bare
         Rugged::Repository.clone_at parent.satelliterepopath, self.satelliterepopath
       end
-      self.pushtobare
+      self.pushtobare unless satelliterepo.empty?
```

Comment:
This fixes 3 of the tests, we don't want to push something to a bare when it's already empty.